h/static/scripts/controllers/form-controller.js

```diff
@@ -243,7 +243,7 @@ class FormController extends Controller {
         // If there was an error processing the request or the server could
         // not be reached, display a helpful error
         this.setState({
-          submitError: err.reason,
+          submitError: err.reason || 'Form Processing Error',
```

Comment:
sure though I was trying to match the tone of the other errors "Form submission failed" "Form validation failed" etc. But I am fine with what you suggested. I will update and merge